specs/form-associated-custom-element.md

```diff
@@ -0,0 +1,114 @@
+  1. any parent `<label>`element. Can be retrieved from the proxy element's `labels` property
+  2. any `<label>` element with `[for="$id"]`, where $id is the `id` attribute *of the custom element*. Because the custom element does not reflect it's `id` attribute to the proxy element (that would result in two elements with the same ID in the DOM), the labels are not automatically associated to the `labels` property of the proxy.
+
+`NodeList`s are not constructable in JavaScript, and because we need to construct the label set in cases where we're using a proxy element, I'm purposing standardizing this property to `Node[]` , which AFAIK is as close to `NodeList` as we can get.
+
+#### Clicking labels
+Clicking a label of a native input element can have several side effects. First (and in general) it will focus the element. In cases like radio and checkbox, it changes the value of the input. There are likely other side-effects.
+
+How this works under the hood seems to be that clicking the label *also* invokes a click event on the element the label *labels*.
+
+However, because (in proxy element cases) the label's aren't *actually* associated to the element, the click event on the custom element isn't ever fired. This is the case for all label's using the `for` attribute to make the association. *Wrapping* labels (`<label><custom-element></label>`) *do* seem to fire the click event on the custom element.
+
+In order to *truly* support this scenario, we need to attach click event handlers to all labels for the elements. It *also* means that we need to attach new listeners when *new* labels get added to the DOM. This is the tricky part. We could use mutation observer, but my fear is that is heavy-handed for an edge-case scenario. 
+
+Another options would be to document the gap and not support this scenario, instead only attaching listeners to elements that are in the DOM when the component connects. Any other thoughts here?
+
+### Accessibility
+
+No accessibility concerns for this utility. Accessibility will be the concern of the implementing class.
+
+### Globalization
+
+No globalization concerns.
+
+### Security
+
+No new security concerns. Form data sanitation will be the responsibility of the consuming application.
+
+### Performance
+
+Potentially using a MutationObserver to attach new click handlers to label elements. TBD on if we want to do this.
+
+### Dependencies
+
+No Dependencies
+
+### Test Plan
+
+TBD. Form association APIs are very new so JSDOM is not likely to expose the feature. It might make sense to do in-browser tests against browsers both with and without support to ensure parity.
+
+
+## Resources
+- https://docs.google.com/document/d/1JO8puctCSpW-ZYGU8lF-h4FWRIDQNDVexzHoOQ2iQmY/edit?pli=1
+- https://html.spec.whatwg.org/multipage/custom-elements.html#custom-elements-face-example
+- https://web.dev/more-capable-form-controls/
```

Comment:
Can these be friendly named links?

Reply:
what should the friendly name be?

Reply:
Maybe just their titles, so first "Form Participation API Explained", second "Creating a form-associated custom element", third "More capable form controls".